airbyte-webapp/src/packages/cloud/components/experiments/SignupSourceDropdown/useGetSourceDefinitions.tsx

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from "react-query";
+
+import { DestinationDefinitionRead, SourceDefinitionRead } from "core/request/AirbyteClient";
+
+import availableSourceDefinitions from "./sourceDefinitions.json";
+
+interface Catalog {
+  destinations: DestinationDefinitionRead[];
+  sources: SourceDefinitionRead[];
+}
+const fetchCatalog = async (): Promise<Catalog> => {
+  const path = "https://storage.googleapis.com/prod-airbyte-cloud-connector-metadata-service/cloud_catalog.json";
+  const response = await fetch(path);
+  return response.json();
+};
+export const useGetSourceDefinitions = () => {
+  return useQuery<Catalog, Error, Catalog["sources"]>("cloud_catalog", fetchCatalog, {
+    select: (data) => {
+      return data.sources.map((source) => {
```

Comment:
This catalog contains more connectors than there are actually availble on cloud. We have some connectors that are deployed on cloud, and are working in existing connections but are not available for creating new connections with (for different reasons). We filter them out in the frontend, and this code would also need to make sure it's filtering them out. The function for this is: https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/core/domain/connector/constants.ts#L18